data/posts/rsc-framework-mode-preview.md

```diff
@@ -0,0 +1,117 @@
+```
+
+## Framework Mode vs Data Mode
+
+In case you're unfamiliar with React Router's Framework Mode and Data Mode, here's a quick overview.
+
+Data Mode provides the core APIs for routing, data loading and action handling. In this mode, you're responsible for the overall structure of your project, potentially managing multiple builds and the communication between them at runtime, wiring everything up to a bundler including HMR etc.
+
+All of this can be tricky to get right. You're effectively building your own meta-framework on top of React Router.
+
+With Framework Mode, all of this build setup is handled for you via our official Vite plugin, including additional features like typegen and optional file system routing. For most people starting a new project, Framework Mode provides the most streamlined developer experience.
+
+So, while many consumers were excited to see RSC support in Data Mode, we know that many more of you were instead patiently waiting for RSC support to land in Framework Mode.
+
+## Layering on Framework Mode
+
+When introducing RSC to React Router, it was important that we first started at the lowest level with RSC Data Mode. Not only did this ensure we have a stable foundation that could be used with any RSC-enabled bundler, but it also gave us a clear compilation target for RSC Framework Mode.
+
+Now that RSC Data Mode is in place, we're able to provide an RSC-powered Vite plugin that compiles to these lower level APIs.
+
+As noted in our post on ["React Router and React Server Components: The Path Forward,"](./react-router-and-react-server-components) the great thing about RSC is that this new Framework Mode plugin is much simpler than our earlier non-RSC work. Most of the framework-level complexity is now implemented at a lower level in RSC Data Mode, with RSC Framework Mode being a more lightweight layer on top.
+
+## Getting started
+
+Today, when using the regular non-RSC Framework Mode, you add the React Router plugin to your Vite config:
+
+<!-- prettier-ignore -->
+```ts
+// vite.config.ts
+import { defineConfig } from "vite";
+import { reactRouter } from "@react-router/dev/vite";
+
+export default defineConfig({
+  plugins: [
+    reactRouter(),
+  ],
+});
+```
+
+To enable RSC Framework Mode, you simply swap this out for the new `unstable_reactRouterRSC` Vite plugin, along with the official (experimental) [@vitejs/plugin-rsc](https://www.npmjs.com/package/@vitejs/plugin-rsc) as a peer dependency.
+
+```sh
+npm install @vitejs/plugin-rsc
+```
+
+<!-- prettier-ignore -->
+```ts
+// vite.config.ts
+import { defineConfig } from "vite";
+import { unstable_reactRouterRSC } from "@react-router/dev/vite";
+import { rsc } from "@vitejs/plugin-react";
+
+export default defineConfig({
+  plugins: [
+    unstable_reactRouterRSC(),
+    rsc(),
+  ],
+});
+```
+
+With these new plugins, all of the features described in our [RSC Data Mode preview blog post](./rsc-preview) will be available in your Framework Mode app:
```

Comment:
Similar to my above comment, I think this is the right blog post but that post isn't about "Data Mode", so I don't think it's the right mention

```suggestion
With these new plugins, all of the features described in our [RSC preview blog post](./rsc-preview) will be available in your Framework Mode app:
```